junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/execution/injection/sample/TypeBasedMapOfListsParameterResolver.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2015-2019 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.engine.execution.injection.sample;
+
+import static java.util.Arrays.asList;
+
+import java.util.List;
+import java.util.Map;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.engine.support.TypeBasedParameterResolver;
+
+/**
+ * @since 5.6
+ */
+public class TypeBasedMapOfListsParameterResolver extends TypeBasedParameterResolver<Map<String, List<Integer>>> {
+	@Override
+	protected Map<String, List<Integer>> resolve(ParameterContext parameterContext, ExtensionContext extensionContext)
```

Comment:
Having `Map<String, List<Integer>>` as the return type instead of `Object` is better, imho. The compiler will bail out of you return `"123"` or any object that's not an instance of `T` here.

Reply:
Agreed.